### ISSUE TITLE
numpyro equivalent to Pyro's poutine.mask

### ISSUE THREAD

**Issue by FlorianWilhelm** [pyro-ppl/numpyro#506]
For me, it seems that numpyro's `primitives` module is missing right now the functionality of Pyro's [poutine.mask](http://docs.pyro.ai/en/stable/poutine.html#pyro.poutine.handlers.mask) in order to mask out some of the sample statements elementwise. This is very convenient when dealing with missing values or if some padding is needed to align for instance time series.

**Comment by neerajprad** [comment]
Thanks for creating this feature request. We usually wait for internal use cases and user requests to prioritize new features. We also have a use case for `poutine.mask`, and we'll plan to have it in the next release.

**Comment by VincentBt** [comment]
Also very interested!
To do things like
```
z_t = pyro.sample("z_%d" % t,
                  dist.Normal(z_loc, z_scale)
                      .mask(mini_batch_mask[:, t - 1:t])
                      .to_event(1))
```

**Comment by fehiepsi** [comment]
This is addressed in #515. For masked distribution, please refer to #520.
